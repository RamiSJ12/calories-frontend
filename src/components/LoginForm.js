import {
  Button,
  FormControl,
  Input,
  InputLabel,
  makeStyles,
} from '@material-ui/core';
import React from 'react';
import PropTypes from 'prop-types';
import commonStyles from '../utils/commonStyles';

const useStyles = makeStyles({
  root: {
    '& input': {
      color: 'black',
    },
  },
});

const LoginForm = ({ submitHandler, changeHandler, state }) => {
  const commonClasses = commonStyles();
  const classes = useStyles();

  return (
    <form className={`${commonClasses.centerVertically} ${commonClasses.topMargin40} ${commonClasses.bottomMargin50}`}>
      <FormControl className={`${commonClasses.topMargin15} ${commonClasses.width80}`}>
        <InputLabel htmlFor="email">Email</InputLabel>
        <Input
          className={`${commonClasses.darkText} ${classes.root}`}
          value={state.email}
          name="email"
          onChange={e => changeHandler(e)}
          id="email"
          type="email"
          required
        />
      </FormControl>
      <FormControl className={`${commonClasses.topMargin15} ${commonClasses.width80}`}>
        <InputLabel htmlFor="password">Password</InputLabel>
        <Input
          className={`${commonClasses.darkText} ${classes.root}`}
          value={state.password}
          name="password"
          onChange={e => changeHandler(e)}
          id="password"
          type="password"
          required
        />
      </FormControl>
      <Button
        type="submit"
        onClick={e => submitHandler(e, state)}
        className={`${commonClasses.topMargin30} ${commonClasses.whiteText}`}
        variant="contained"
        color="primary"
      >
        LOGIN
      </Button>
    </form>
  );
};

LoginForm.propTypes = {
  submitHandler: PropTypes.func.isRequired,
  changeHandler: PropTypes.func.isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  state: PropTypes.object.isRequired,
};

export default LoginForm;
