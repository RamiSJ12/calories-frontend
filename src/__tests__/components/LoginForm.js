import renderer from 'react-test-renderer';
import LoginForm from '../../components/LoginForm';

describe('LoginForm component', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<LoginForm
      state={{}}
      submitHandler={() => ({})}
      changeHandler={() => ({})}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
