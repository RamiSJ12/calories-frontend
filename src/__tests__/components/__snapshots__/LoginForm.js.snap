// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component should match snapshot 1`] = `
<form
  className="makeStyles-centerVertically-6 makeStyles-topMargin40-7 makeStyles-bottomMargin50-8"
>
  <div
    className="MuiFormControl-root makeStyles-topMargin15-10 makeStyles-width80-22"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink={false}
      htmlFor="email"
    >
      Email
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-darkText-3 makeStyles-root-24 MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
      required={true}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        id="email"
        name="email"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="email"
      />
    </div>
  </div>
  <div
    className="MuiFormControl-root makeStyles-topMargin15-10 makeStyles-width80-22"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink={false}
      htmlFor="password"
    >
      Password
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-darkText-3 makeStyles-root-24 MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
      required={true}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        id="password"
        name="password"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="password"
      />
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-topMargin30-11 makeStyles-whiteText-12 MuiButton-containedPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="submit"
  >
    <span
      className="MuiButton-label"
    >
      LOGIN
    </span>
  </button>
</form>
`;
